_inc/search/src/index.jsx

```diff
@@ -0,0 +1,30 @@
+/** @jsx h */
+
+/**
+ * External dependencies
+ */
+import { h, render } from 'preact';
+
+/**
+ * Internal dependencies
+ */
+import JetpackSearchAPI from '../components/api';
+import SearchResults from '../components/search-results';
+import SearchWidget from '../components/search-widget';
+
+const api = new JetpackSearchAPI();
+
+const injectSearchWidget = ( initialValue, target, grabFocus ) => {
+	render(
+		<SearchWidget
+			initialValue={ initialValue }
+			grabFocus={ grabFocus }
```

Comment:
What should happen when there are multiple widgets requesting to `grabFocus`?

Reply:
Whichever fires off last would grab the focus. But, that's up to the higher level application to manage. `grabFocus` is there so if we're using progressive enhancement, we can maintain the focus on the search field if the user already has the search field focused. If the page is saying there are multiple input fields with focus at the same time, then the logic probably needs fixing there :)